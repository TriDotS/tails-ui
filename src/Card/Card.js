import React from 'react'
import styled from 'styled-components'

const Card = styled.div`
  padding: 5px;
  border: ${props => (props.bordered ? '1px solid silver' : 'none')};
  border-radius: ${props => (props.rounded ? '3px' : '0px')};
  background: ${props => props.backgroundColor || 'white'};
  box-shadow: ${props =>
    props.shadowed ? '0 4px 8px 0 rgba(0,0,0,0.2)' : null};
  color: ${props => props.fontColor || 'black'};
`

const CardHeader = styled.div`
  font-weight: bolder;
  display: flex;
  align-items: center;
  font-size: 1.5em;
  color: ${props => props.headerColor || 'black'};
  border-bottom: 1px solid silver;
  margin-bottom: 5px;
  padding: 5px;
`

const CardContent = styled.div`
  padding: 5px;
  font-size: 1em;
`

export default ({
  children,
  headerText,
  shadowed,
  bordered,
  rounded,
  backgroundColor,
  fontColor,
  headerColor,
}) =>
  <Card
    shadowed={shadowed}
    bordered={bordered}
    rounded={rounded}
    backgroundColor={backgroundColor}
    fontColor={fontColor}
  >
    {headerText &&
      <CardHeader headerColor={headerColor}>{headerText}</CardHeader>}
    <CardContent>
      {children}
    </CardContent>
  </Card>
