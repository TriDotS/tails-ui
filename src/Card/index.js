import React from 'react'
import styled from 'styled-components'

const Card = styled.div`
  padding: 5px;
  border: 1px solid silver;
  border-radius: 3px;
  background: white;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
`

const CardHeader = styled.div`
  font-weight: bolder;
  display: flex;
  align-items: center;
  font-size: 2em;
  color: darkslategrey;
  border-bottom: 1px solid silver;
  margin-bottom: 5px;
  padding: 5px;
`

const CardContent = styled.div`
  padding: 5px;
  font-size: 1em;
`

export default ({ children, headerText }) =>
  <Card>
    {headerText && <CardHeader>{headerText}</CardHeader>}
    <CardContent>
      {children}
    </CardContent>
  </Card>
