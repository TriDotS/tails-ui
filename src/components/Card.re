let component = ReasonReact.statelessComponent("Card");

let make =
    (
      ~title: string,
      ~footer: array(ReasonReact.reactElement),
      ~children: array(ReasonReact.reactElement)
    ) => {
  ...component,
  render: _self =>
    <div className="max-w-sm rounded shadow-lg">
      <div className="px-6 py-4">
        <div className="font-bold text-xl mb-4">
          (ReasonReact.stringToElement(title))
        </div>
        (ReasonReact.arrayToElement(children))
      </div>
      <div className="px-6 py-4"> (ReasonReact.arrayToElement(footer)) </div>
    </div>
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~title=jsProps##title,
      ~footer=jsProps##footer,
      ~children=jsProps##children
    )
  );
