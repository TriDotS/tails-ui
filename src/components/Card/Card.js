import PropTypes from 'prop-types'
import React from 'react'

const Card = ({ title, children, footer }) => (
  <div className="max-w-sm rounded overflow-hidden shadow-lg">
    <div className="px-6 py-4">
      <div className="font-bold text-xl mb-4">{title}</div>
      {children}
    </div>
    <div className="px-6 py-4">{footer}</div>
  </div>
)

Card.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node,
  footer: PropTypes.node
}

export default Card
