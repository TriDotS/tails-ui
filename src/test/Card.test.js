import Card from '../components/Card'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(
    <Card title="Card" footer={<a href="#!">footer link</a>}>
      This is a card
    </Card>,
    div
  )
})

test('card snapshot', () => {
  const tree = renderer
    .create(
      <Card title="Card" footer={<a href="#!">footer link</a>}>
        This is a card
      </Card>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
