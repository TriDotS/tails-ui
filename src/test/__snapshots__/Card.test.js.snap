// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card snapshot 1`] = `
<div
  className="max-w-sm rounded overflow-hidden shadow-lg"
>
  <div
    className="px-6 py-4"
  >
    <div
      className="font-bold text-xl mb-4"
    >
      Card
    </div>
    This is a card
  </div>
  <div
    className="px-6 py-4"
  >
    <a
      href="#!"
    >
      footer link
    </a>
  </div>
</div>
`;
